### PR TITLE
hyperopt: -j/--job-workers command line option added

### PR DESCRIPTION
**Pull request by hroff-1902**
A part of #1775.

This option allows to control the number of the joblib.parallel worker processes used in the hyperopt.

If set to -1 (default), all CPUs are used (joblib parallel backend determines the number of the worker processes automatically), for -2, all CPUs but one are used, etc. If 1 is given, no parallel computing code is used at all.

Needs to be tested additionally on Windows and on VPS or virtual machines with 1 CPU. Please verify the solution in these conditions before merge.

* Docs adjusted as well. No tests added.
* As an additional minor cosmetics, redundant comments that do not apply any additional useful information but require attention while copying/extending options were removed from freqtrade/configuration.py

### CODE REVIEW

File: freqtrade/optimize/hyperopt.py [hunk 1]
```diff
@@ -270,21 +270,25 @@ def start(self) -> None:
 
         cpus = multiprocessing.cpu_count()
         logger.info(f'Found {cpus} CPU cores. Let\'s make them scream!')
```
[the remark this answers is not in the document]
Comment (hroff-1902) [reply freqtrade/optimize/hyperopt.py]: I removed dependency on multiprocessing, it's no longer imported... 😎 

Let's leave this message -- "for historical reasons", I like it...

**Comment by hroff-1902** [comment]
Need help, cannot understand why the test fail.
If it worked with `freqtrade.optimize.hyperopt.multiprocessing.cpu_count` why does not it work with `freqtrade.optimize.hyperopt.joblib.cpu_count`?

**Comment by hroff-1902** [comment]
Moreover, the pytest does not fail running locally (git status is up to date)... Hmm...

File: freqtrade/optimize/hyperopt.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (xmatthias) [reply freqtrade/optimize/hyperopt.py]: Even better!

**Comment by xmatthias** [comment]
it does fail on my system...

the mock is missing the module - multiprocessing was imported completely, while we import specific functions from joblib.
try the following:
 
``` python
mocker.patch('freqtrade.optimize.hyperopt.cpu_count', MagicMock(return_value=1))
```

However, it'll fail nonetheless, since we mock the results (and only pass in 1 result) - so for this test you either need to mock more results - or pass in -j 1 (which was the behaviour before on this test).

**Comment by hroff-1902** [comment]
> Also look out for line 301 - it'll want to be jobs, not cpu's ... (accidentally noticed this ...)

Right 👍 It left in this state because of multiple merges already done yesterday.

**Pull request merged**